source/AI.cpp

```diff
@@ -2861,22 +2861,38 @@ void AI::AimTurrets(const Ship &ship, FireCommand &command, bool opportunistic)
 				// have moved forward one time step.
 				p += v;
 
-				// Find out how long it would take for this projectile to reach the target.
-				double rendezvousTime = RendezvousTime(p, v, vp);
-				// If there is no intersection (i.e. the turret is not facing the target),
-				// consider this target "out-of-range" but still targetable.
-				if(std::isnan(rendezvousTime))
-					rendezvousTime = max(p.Length() / (vp ? vp : 1.), 2 * weapon->TotalLifetime());
+				double rendezvousTime;
+				// Beam weapons hit instantenously if they are in range.
+				if(weapon->TotalLifetime() == 1)
+				{
+					// Lower target priority if it is out of range.
+					double distance = p.Length();
+					if(distance > weapon->Velocity())
+						rendezvousTime = distance / weapon->Velocity() - 1.;
+					else
+						rendezvousTime = 0.;
+				}
+				else
+				{
+					// Find out how long it would take for this projectile to reach the target.
+					rendezvousTime = weapon->TotalLifetime() == 1 ? 0 : RendezvousTime(p, v, vp);
```

Comment:
This bit will never run.
```suggestion
					rendezvousTime = RendezvousTime(p, v, vp);
```